src/qml/filters/fisheye/ui.qml

```diff
@@ -0,0 +1,827 @@
+        }
+        Shotcut.SliderSpinner {
+            visible: stretchShowSlider
+            id: stretchSlider
+            minimumValue: -0.5
+            maximumValue: 0.5
+            decimals: 2
+            suffix: ''
+            value: 0
+            onValueChanged: { if (blockUpdate) return; filter.set(stretchParam, value+0.5) }
+        }
+        Shotcut.UndoButton {
+            visible: stretchShowSlider
+            onClicked: stretchSlider.value = stretchDefault - 0.5
+        }
+
+
+        //line seperator #1
+         RowLayout {
+            Layout.columnSpan: 3
+            Label { text: qsTr('Scale')}  
+            Rectangle{ Layout.fillWidth: true;height: 1; color: activePalette.text; opacity: 0.3 }
+        }
+        
+        // Row 7: Scale (Preset)
+        Label {
+            text: qsTr('Presets')
+            Shotcut.HoverTip { text: qsTr( 'Preset scale methods\n'+
+                                           'Lock pixels at specific locations') }
+            Layout.alignment: Qt.AlignRight
+        }
+        RowLayout {
+            Shotcut.ComboBox {
+                id: scaleCombo
+                implicitWidth: 120
+                model: ListModel {
+                    id: scaleModel
+                    ListElement { text: qsTr('1. Scale to Fill'); value: 0.0 }
+                    ListElement { text: qsTr('2. Keep Center Scale'); value: 0.333 }
+                    ListElement { text: qsTr('3. Scale to Fit'); value: 0.666 }
+                    ListElement { text: qsTr('4. Manual Scale'); value: 1.0 }
+                }
+                textRole: 'text'
+                onActivated: {
+                    filter.set(scaePresetParam, scaleModel.get(currentIndex).value)
+                    scaleShowSlider = ((currentIndex+1)==scaleCombo.count) //show user input?
+                    setScollbarHeight()
+                }
+            }
+            CheckBox {
+                text: qsTr('Y')
+                leftPadding: 6
+                Shotcut.HoverTip { text: qsTr('Scale Y separately\nThis changes video aspect ratio') }
+                id: scaleYCheckBox
+                padding: 0
+                checked: false
+                onCheckedChanged: {
+                    if (blockUpdate) return
+                    filter.set(scaleYParam, checked? scaleYSlider.value+0.5:0.5)
+                    scaleYShowSlider = checked                  
+                    setScollbarHeight()
+                }
+            }            
+            CheckBox {
+                text: qsTr('Crop')
+                leftPadding: 6
+                Shotcut.HoverTip { text: qsTr('Remove distorted edges') }                
+                id: cropCheckBox
+                padding: 0
+                checked: cropDefault
+                visible: fisheyeRemoveButton.checked
+                onCheckedChanged: { if (blockUpdate) return; filter.set(cropParam, checked) }
+            }

```

Comment:
We need a way to hide the Crop, Stretch, and Y ratio UI elements when the frei0r filter is not recent enough. There are Linux and BSD distributions that only include release frei0r plugins, and those are only released around every 2 years! A simple way to do this is by checking the presence of a new non-zero (due to how MLT works) default parameter value from the frei0r plugin. For example, the new property '10' has a frei0r/MLT default value of 1.0. In older versions of this plugin, MLT returns 0.0 when this property is requested. So, before defaults are set in `Component.onCompleted` check if this property value is > 0 and set a boolean QML property that you refer to in this object's `visible` property. You do not need to worry about using unavailable properties in the rest of the logic except where it will lead to a divide by zero. If you request the value of a non-existing parameter with `filter.getDouble()` it returns 0.